recipe/recipe.yaml

```diff
@@ -0,0 +1,181 @@
+schema_version: 1
+
+context:
+  version: 3.22.3
+  build: 101
+  # these variables should all be overridden in variant files
+  # but default values are sometimes needed for conda-smithy rerender
+  mpi: ${{ mpi | default('mpich') }}
+  scalar: ${{ scalar | default('real') }}
+  cuda_compiler_version: ${{ cuda_compiler_version | default('None') }}
+  # construct some variables to use in the recipe
+  # with_cuda can be used as a bool for cuda or not conditions
+  with_cuda: ${{ 'true' if cuda_compiler_version != 'None' else '' }}
+  # cuda_major is always the major version of cuda as an integer string ('0' if not with_cuda)
+  cuda_major: >-
+    ${{
+      cuda_compiler_version | split('.') | first
+      if with_cuda
+      else "0"
+    }}
+  # construct the build prefix (e.g. 'cuda12_complex' or 'real')
+  cuda_prefix: ${{ "cuda" + cuda_major + "_" }}
+  build_prefix: ${{ cuda_prefix + scalar if with_cuda else scalar }}
+
+package:
+  name: petsc
+  version: ${{ version }}
+
+source:
+  url: https://web.cels.anl.gov/projects/petsc/download/release-snapshots/petsc-${{ version }}.tar.gz
+  sha256: 88c0d465a3bd688cb17ebf06a17c06d6e9cc457fa6b9643d217389424e6bd795
+  patches:
+    - ignore-not-invalid.patch
+    - no-cppflags-in-pkgconfig-cflags.patch
+    - target-cudadir.patch
+
+build:
+  number: ${{ build }}
+  string: ${{ build_prefix }}_h${{ hash }}_${{ build }}
+  skip: win
+  variant:
+    # deprioritize cuda and complex with track_features
+    down_prioritize_variant: >
+      ${{
+        (1 if with_cuda else 0)
+      + (1 if scalar == "complex" else 0)
+      }}
```

Comment:
This syntax may be causing an issue due to the fact that it is changing the interpretation of `track_features`
https://github.com/conda-forge/conda-forge.github.io/issues/2437#issuecomment-2628979198

Personally, I've found `track_features` to be rather inconsistent and simply used build number to prioritize things. Its a little messy, 

I'm trying to find where the finalized recipe is, but I guess it isn't the same thing with v1 recipes.

cc: @wolfv

Reply:
Yes this seems like a bug in rattler/rattler-build. Sorry

Reply:
Adding a trackable record, there's nothing wrong with the syntax here, it's purely a small bug in rattler's serialization of track_features when `down_prioritize_variant >= 2` (why it only affects `cuda + complex` builds), fixed in https://github.com/conda/rattler/pull/1039